src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
And actually looking more closely at the definition of `%`...

The existence of `x % y` implies:

    * the existence of `x - x`, and
    * the existence of `x / y`, and
    * the existence of `(x/y) * y`

where `x` and `y` may or may not be the same type.  On the third bullet, `(x/y)` means the type of that result must be able to be multiplied by the type of `y`.  The type of `x/y` may not be the same type as `x` and `y`.  For example in `<chrono>` when `x` is `seconds` and `y` is milliseconds, `x/y` is `int`.  That being said, I may be stretching the analogy of `chrono` and `tagged_integer` too far...